pkg/deploy/lattice/service_manager.go

```diff
@@ -187,7 +189,16 @@ func (m *defaultServiceManager) updateAssociations(ctx context.Context, svc *Ser
 	}
 
 	for _, assoc := range toDelete {
-		m.deleteAssociation(ctx, assoc.Arn)
+		isManaged, err := m.cloud.IsArnManaged(*assoc.Arn)
```

Comment:
not for now, but in future we can also optimize by avoiding making a call to get tags when createdBy is different account for association.